qiime2/tests/test_metadata.py

```diff
@@ -36,6 +37,30 @@ def test_does_not_cast_index_or_column_types(self):
             check_exact=True)
 
 
+class TestMetadataFromArtifact(unittest.TestCase):
+    def setUp(self):
+        get_dummy_plugin()
+
+    def test_from_artifact(self):
+        A = qiime2.Artifact.import_data('Mapping', {'a': '1', 'b': '3'})
+        md = qiime2.Metadata.from_artifact(A)
+        pdt.assert_frame_equal(md.to_dataframe(),
+                               pd.DataFrame({'a': '1', 'b': '3'}, index=['0']))
+
+    def test_from_bad_artifact(self):
+        A = qiime2.Artifact.import_data('IntSequence1', [1, 2, 3, 4])
+        with self.assertRaisesRegex(ValueError, 'metadata'):
+            qiime2.Metadata.from_artifact(A)
+
+    def test_artifact(self):
+        A = qiime2.Artifact.import_data('Mapping', {'a': ['1', '2'],
+                                                    'b': ['2', '3']})
+        md = qiime2.Metadata.from_artifact(A)
+        art = md.artifact
+        self.assertIsInstance(art, qiime2.Artifact)
+        self.assertEquals(art, A)
```

Comment:
Can you use `self.assertEqual`? `assertEquals` is [deprecated](https://docs.python.org/3/library/unittest.html#deprecated-aliases).